File: Interviews/GFG/random-problems/Medium/Find-patterns/Find-patterns.cpp
class Solution {
   public:
    int numberOfSubsequences(string S, string W) {
        int i = 0, ans = 0;

        while (i < S.size()) {
            int k = 0;
            for (int j = i; j < S.size(); j++) {
                if (W[k] == S[j]) {
                    if (k == 0)
                        i = j + 1;

                    k++;
                    S[j] = '-1';
                }
                if (k == W.size())
                    break;
            }

            if (k == W.size())
                ans++;
            else
                break;
        }

        return ans;
    }
};

File: Interviews/GFG/random-problems/Medium/Find-patterns/Find-patterns.py
class Solution:
    def numberOfSubsequences(self, S, W):
        i, ans = 0, 0

        temp = list(S)
        while i < len(temp):
            k = 0
            for j in range(i, len(temp)):
                if W[k] == temp[j]:
                    if k == 0:
                        i = j + 1
                    k += 1
                    temp[j] = '-1'
                if k == len(W):
                    break
            if k == len(W):
                ans += 1
            else:
                break

        return ans
